From 067e5a8a4931dd9208394e05d820e9fce5a7a6b8 Mon Sep 17 00:00:00 2001
From: Guoqing Jiang <guoqing.jiang@suse.com>
Date: Sat, 7 May 2022 15:37:06 +0800
Subject: [PATCH] Revert "virtio_net: Fix error handling in virtnet_restore()"
Patch-mainline: Never, revert stable patch which is added again in sorted section
References: bsn#198
Modified-by-SEL: No


This reverts commit 09a94a89d74fcd57bea948d19f08f602c368ac99.

Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 drivers/net/virtio_net.c | 5 +----
 1 file changed, 1 insertion(+), 4 deletions(-)

diff --git a/drivers/net/virtio_net.c b/drivers/net/virtio_net.c
index 345a0f51e8d7..91e0e6254a01 100644
--- a/drivers/net/virtio_net.c
+++ b/drivers/net/virtio_net.c
@@ -3234,11 +3234,8 @@ static __maybe_unused int virtnet_restore(struct virtio_device *vdev)
 	virtnet_set_queues(vi, vi->curr_queue_pairs);
 
 	err = virtnet_cpu_notif_add(vi);
-	if (err) {
-		virtnet_freeze_down(vdev);
-		remove_vq_common(vi);
+	if (err)
 		return err;
-	}
 
 	return 0;
 }
-- 
2.34.1

